lib/pure/asyncdispatch.nim

```diff
@@ -1228,14 +1228,19 @@ else:
     let newLength = max(len(curList), InitCallbackListSize)
     var newList = newSeqOfCap[Callback](newLength)
 
+    var eventsExtinguished = false
     for cb in curList:
+      if eventsExtinguished:
+        newList.add(cb)
+        continue
       if not cb(fd):
         # Callback wants to be called again.
         newList.add(cb)
         # This callback has returned with EAGAIN, so we don't need to
         # call any other callbacks as they are all waiting for the same event
         # on the same fd.
-        break
```

Comment:
It's not sufficient to just remove the `break`? Why is the `eventsExtinguished` flag also necessary?
The `break` is what is causing the other cbs in the curList to not get run. Or are the cb's getting dropped in the next poll cycle as well?

Reply:
oh. I think I understand. It looks like you are pushing all cbs after the triggered one onto the next poll cycle.
Why defer them instead of just continuing the loop and running them in the current poll cycle? (I hope that sentence makes sense. lol)

Reply:
Read the comment:

> This callback has returned with EAGAIN, so we don't need to call any other callbacks as they are all waiting for the same event on the same fd.

Reply:
Ok. Cool. Makes sense. I forgot that this function is for all callbacks waiting for the same fd.

Thanks for the clarification Dom!